WordPress/wporg-main-2022
# Content Parsers: Update parsers, fix list-item parsing, fix broken manifest

This PR updates the parsers using the code added in https://github.com/WordPress/wordpress.org/pull/90, and adds a phpunit env and tests for the parser.

I wanted to fix the list issue #211 so that it didn't require a manual update each time `build:patterns` was run, and I've had this branch half-done for a while in the `wporg-mu-plugins` repo, but it seems faster to iterate on the parsers here.
 
This is probably easier to review going commit-by-commit:

- https://github.com/WordPress/wporg-main-2022/pull/247/commits/12fb450f764e2b9e79dffc0adba1cb3506b618a2 — Update the parsers
- https://github.com/WordPress/wporg-main-2022/pull/247/commits/66e32b0d51767f3a7b27fe8bcec0efa42e0872c6 — Add phpunit environment (copied from #181)
- https://github.com/WordPress/wporg-main-2022/pull/247/commits/e52327d5fe6016ed30d72a066b8bbd5341d9f33a — Remove the broken swag page from the manifest, this was preventing all following pages from exporting
- https://github.com/WordPress/wporg-main-2022/pull/247/commits/88409ceacc9767a1a9e433356a1100bef405ac1e — Run `build:patterns` with the new content
- https://github.com/WordPress/wporg-main-2022/pull/247/commits/f57089a22e8362370a21dad509e9590126a6b04b — Add phpunit step to the github workflows

Fixes #179, closes #181, fixes #211.

Props @adamwoodnz @dd32 for most of the code, I've just moved it around 🙂 

### How to test the changes in this Pull Request:

1. Run `composer update` to get the latest packages
2. Run `yarn build:patterns` to see the parsers still process the patterns
3. Run `yarn test:php` to run the unit tests
